### ISSUE TITLE
[v2,bug] In gallery view with row layout images are vertically cropped

### ISSUE THREAD

**Issue by Artiom-M** [marcantondahmen/automad#103]
As you can see on this picture:
![image](https://github.com/marcantondahmen/automad/assets/4058815/c82f1201-075b-4058-b9b0-f04a707abfe7)
I would expect it to be resized so picture's height matches row height of the gallery.
What I can see is that image is resized correctly, but then expanded:
![image](https://github.com/marcantondahmen/automad/assets/4058815/23690abc-7572-42a9-9a28-928a58817a6c)

**Comment by Artiom-M** [comment]
Just played around and discovered that removing _flex-grow_ is fixing it.
![image](https://github.com/marcantondahmen/automad/assets/4058815/a03f814a-a8e2-41f0-80dc-b4e80f27c4c8)

**Comment by marcantondahmen** [comment]
Hi, this is geometrically impossible since the rows have a fixed height but the width is flexible. There will be always some cropping. However, the render engine tries to minimize the cropping.

**Comment by Artiom-M** [comment]
Yes, if the proportion is so that width will be greater than the content width. However in general case, fixed height is what I want, but in this case it is king of fixed width and cropped height.
So I just want to see the full picture even it is means having bands (vertical or horizontal) around. Something like "fit in" option. Because cropping is often looks ugly, like the bottom left image in my example or middle right. These are both vertical photos...

**Comment by Artiom-M** [comment]
I think I understand were the problem is. The "row" layout tries to fit the row into the page width and so each row has the same width in addition to a fixed height. While "columns" layout allows columns to have different height:

![Image](https://github.com/user-attachments/assets/af7ad289-ebde-4a40-8666-2d6248a699db)

What I am looking for is to have the same for "rows" layout, so the height is the same, and the length of the row depends on how many images will fit.

I think it could be an option for "rows" layout: extend to the page width

**Comment by Artiom-M** [comment]
Will close it because the fix is easy to implement using _Dashboard ->General Data and Files -> Customisations_ : 
```
.am-gallery-flex-item {
    flex-grow: 0;
}
```
